plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -878,23 +878,26 @@ func haveSame(a1, a2 []string) bool {
 
 func GeneralPredicates(pod *v1.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	var predicateFails []algorithm.PredicateFailureReason
-	fit, reasons, err := noncriticalPredicates(pod, meta, nodeInfo)
+	fit, reasons, err := EssentialPredicates(pod, meta, nodeInfo)
```

Comment:
This PR assume `PodSelectorMatches` will return earlier than `PodFitsResources`, so DaemonSet will not report resource error if node selector is enabled. prefer to avoid such kind of assumption.